simplesat/tests/test_repository.py

```diff
@@ -0,0 +1,170 @@
+        dummy_with_proxy_scripts 1.0.0-1
+        dummy_with_proxy_softlink 1.0.0-1
+        nose 1.2.1-1
+        nose 1.3.0-1
+        nose 1.3.0-2\
+        """)
+        packages = self.packages_from_definition(packages_definition)
+
+        # When
+        repository = Repository()
+
+        # Then
+        self.assertEqual(len(repository), 0)
+
+        # When
+        repository = Repository(packages)
+
+        # Then
+        self.assertEqual(len(repository), len(packages))
+        self.assertEqual(list(repository), packages)
+
+        # When
+        repository = Repository(reversed(packages))
+
+        # Then
+        self.assertEqual(len(repository), len(packages))
+        self.assertEqual(list(repository), packages)
+
+    def test_update(self):
+        # Given
+        packages_definition = textwrap.dedent("""\
+        dummy 1.0.1-1
+        dummy_with_appinst 1.0.0-1
+        dummy_with_entry_points 1.0.0-1
+        dummy_with_proxy 1.3.40-3
+        dummy_with_proxy_scripts 1.0.0-1
+        dummy_with_proxy_softlink 1.0.0-1
+        nose 1.2.1-1
+        nose 1.3.0-1
+        nose 1.3.0-2\
+        """)
+        packages = self.packages_from_definition(packages_definition)
+
+        # When
+        repository = Repository(packages[:4])
+
+        # Then
+        self.assertEqual(len(repository), 4)
+
+        # When
+        repository.update(packages[4:])
+
+        # Then
+        self.assertEqual(len(repository), len(packages))
+        self.assertEqual(list(repository), packages)
+
+    def test_find_package(self):
+        # Given
+        packages_definition = textwrap.dedent("""\
+        dummy 1.0.1-1
+        dummy_with_appinst 1.0.0-1
+        dummy_with_entry_points 1.0.0-1
+        dummy_with_proxy 1.3.40-3
+        dummy_with_proxy_scripts 1.0.0-1
+        dummy_with_proxy_softlink 1.0.0-1
+        nose 1.2.1-1
+        nose 1.3.0-1
+        nose 1.3.0-2\
+        """)
+        packages = self.packages_from_definition(packages_definition)
+        repository = Repository(packages)
+
+        # When
+        package = repository.find_package("nose", V("1.3.0-1"))
+
+        # Then
+        self.assertEqual(package.key, "nose-1.3.0-1.egg")
+
+    def test_find_unavailable_package(self):
+        # Given
+        packages_definition = textwrap.dedent("""\
+        dummy 1.0.1-1
+        dummy_with_appinst 1.0.0-1
+        dummy_with_entry_points 1.0.0-1
+        dummy_with_proxy 1.3.40-3
+        dummy_with_proxy_scripts 1.0.0-1
+        dummy_with_proxy_softlink 1.0.0-1
+        nose 1.2.1-1
+        nose 1.3.0-1
+        nose 1.3.0-2\
+        """)
+        packages = self.packages_from_definition(packages_definition)
+        repository = Repository(packages)
+
+        # When/Then
+        with self.assertRaises(NoPackageFound):
+            repository.find_package("nose", V("1.4.0-1"))
+
+        with self.assertRaises(NoPackageFound):
+            repository.find_package("nono", V("1.4.0-1"))
```

Comment:
I think it would be better to choose a version number that exists so that it's clearly the name causing failure.